lib/customisation.nix

```diff
@@ -68,16 +68,17 @@ rec {
     let
       ff = f origArgs;
       overrideWith = newArgs: origArgs // (if lib.isFunction newArgs then newArgs origArgs else newArgs);
+      override = lib.setFunctionArgs (newArgs: makeOverridable f (overrideWith newArgs)) (lib.functionArgs f);
```

Comment:
I think this patch is changing `override` to be a functor instead of a function, in order to carry the arguments with it. 

While I do agree this patch makes sense for `override`, it seems to me that this would not work as when using `override` after an `overrideDerivation` call. The reason being that `makeOverridable` would not be able to infer the list of arguments from the (args: ...) function.

Instead I will suggest to either add a new `makeOverridable` or change the current one to accept an extra `expectedArgs` input, which would be bound to `functionArgs f` by the caller.

In addition, can you check the result of `nox review` to ensure that these changes do not break any packages as the bot seems to imply.

Reply:
> While I do agree this patch makes sense for `override`, it seems to me that this would not work as when using `override` after an `overrideDerivation` call. The reason being that `makeOverridable` would not be able to infer the list of arguments from the (args: ...) function.

I think that's alright, `overrideDerivation` is meant to be lower-level and [`override` is recommended](https://nixos.org/nixpkgs/manual/#sec-pkg-overrideDerivation) in almost all cases.

> Instead I will suggest to either add a new `makeOverridable` or change the current one to accept an extra `expectedArgs` input, which would be bound to `functionArgs f` by the caller.

This almost entirely nullifies the benefits of this change, because you won't be able to query the function arguments for any `.override`, but only the ones that got explicitly enabled for it, which won't happen on a large scale.

I'll see what it takes to fix the evaluation errors

Reply:
I do not consider `overrideDerivation` / `overrideAttrs` to be any lower level than `override`. Both are dealing with making changes at different stages of the evaluation.

Having an argument to recall what are the argument should not make any difference. Why do you think it would make a difference?